src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -149,7 +149,7 @@ public MsQuicConnection(IPEndPoint localEndPoint, IPEndPoint remoteEndPoint, MsQ
 
             try
             {
-                Debug.Assert(!Monitor.IsEntered(_state));
+                Debug.Assert(!Monitor.IsEntered(_state), "!Monitor.IsEntered(_state)");
```

Comment:
What led you to even expand more the messages in asserts after Stephen's comment?
It'll become redundant.

Reply:
I thought it would be more useful to have a message handy when something crashes, I thought we generally tend to include the message with the assert in this repo

Reply:
It's exactly the other way around, just grep for `Debug.Assert` in the repo. And if it contains the message, I mostly see some additional info and not just the copy of the condition.